frontend/src/components/MapView/DateSelector/TimelineItems/index.tsx

```diff
@@ -37,56 +40,61 @@ const TimelineItems = memo(
 
     const { t } = useSafeTranslation();
 
+    const selectedLayerDateItems: DateItem[][] = useMemo(
+      () =>
+        selectedLayers.map(layer => {
+          return (layer as DateCompatibleLayerWithDateItems).dateItems;
+        }),
+      [selectedLayers],
+    );
+
     // Hard coded styling for date items (first, second, and third layers)
     const DATE_ITEM_STYLING: {
       class: string;
       color: string;
-    }[] = useMemo(() => {
-      return [
-        { class: classes.intersectionDate, color: 'White' },
-        { class: classes.layerOneDate, color: 'Blue' },
-        { class: classes.layerTwoDate, color: 'Yellow' },
-        { class: classes.layerThreeDate, color: 'Red' },
-        // For now, super-impose additional layers in case we have too many.
-        // TODO - handle this more cleanly.
-        { class: classes.layerThreeDate, color: 'Blue' },
-        { class: classes.layerThreeDate, color: 'Yellow' },
-      ];
-    }, [
-      classes.intersectionDate,
-      classes.layerOneDate,
-      classes.layerThreeDate,
-      classes.layerTwoDate,
-    ]);
-
-    const formattedSelectedLayerDates = useMemo(
-      () =>
-        selectedLayerDates.map(layerDates =>
-          layerDates.map(layerDate => formatDate(layerDate)),
-        ),
-      [selectedLayerDates],
-    );
+    }[] = [
+      { class: classes.intersectionDate, color: 'White' },
+      { class: classes.layerOneDate, color: 'Blue' },
+      { class: classes.layerTwoDate, color: 'Yellow' },
+      { class: classes.layerThreeDate, color: 'Red' },
+      // For now, super-impose additional layers in case we have too many.
+      // TODO - handle this more cleanly.
+      { class: classes.layerThreeDate, color: 'Blue' },
+      { class: classes.layerThreeDate, color: 'Yellow' },
+    ];
 
-    const formattedIntersectionDates = useMemo(
-      () => intersectionDates.map(layerDate => formatDate(layerDate)),
-      [intersectionDates],
-    );
+    const DIRECTION_ITEM_STYLING: {
+      class: string;
+    }[] = [
+      {
+        class: classes.layerOneDirection,
```

Comment:
How about layerThree?
Maybe you could include this in the mapping above already as directionClass or something like that?

Reply:
Is it possible to have 3 overlaping layers ? I thought you said that 2 was the maximum ?

Reply:
Also, it relies on the index of the array, so I don't see a simple way to include it in the above array

Reply:
Yes you can have 3 layers currently if you overlap raster (wms), admin level data, and point data for example, sorry if I mislead you.

You could use an array of objects then. Not sure I understand where the problem is atm.

Reply:
To test for 3 dated layers, you can use southsudan and these layers:

http://localhost:3000/?hazardLayerIds=flood_events%2Crainfall_dekad%2Cconflict_fatalities&date=2022-12-11

Reply:
Thanks, I will drive additional tests on that !

Reply:
Glad you gave me that example ! I found that I had issue with discontinuous data points. We are OK now